lib/phoenix_component.ex

```diff
@@ -1310,10 +1347,22 @@ defmodule Phoenix.Component do
     raise_bad_socket_or_assign!("changed?/2", assigns)
   end
 
+  defmacro embed_templates(pattern, opts \\ []) do
+    quote do
+        Phoenix.Template.compile_all(
+          &Phoenix.Component.__embed__/1,
+          unquote(opts)[:root] || __DIR__,
+          unquote(pattern)
+        )
+    end
+  end
```

Comment:
Extra `end`, sorry. We also don't need before_compile or @phoenix_embeds anymore I think. :)